include/sentry.h

```diff
@@ -1553,6 +1564,30 @@ SENTRY_EXPERIMENTAL_API void sentry_span_set_status(
 SENTRY_EXPERIMENTAL_API void sentry_transaction_set_status(
     sentry_transaction_t *tx, sentry_span_status_t status);
 
+/**
+ * Type of the `iter_headers` callback.
+ *
+ * The callback is being called with HTTP header key/value pairs.
+ * These headers can be attached to outgoing HTTP requests to propagate
+ * distributed tracing metadata to downstream services.
+ *
+ */
+typedef void (*sentry_iter_headers_function_t)(
+    const char *key, const char *value, void *userdata);
+
+/**
+ * Iterates the distributed tracing HTTP headers for the given span.
+ */
+SENTRY_EXPERIMENTAL_API void sentry_span_iter_headers(sentry_span_t *span,
+    sentry_iter_headers_function_t callback, void *userdata);
+
+/**
+ * Iterates the distributed tracing HTTP headers for the given transaction.
+ */
+SENTRY_EXPERIMENTAL_API void sentry_transaction_iter_headers(
+    sentry_transaction_t *tx, sentry_iter_headers_function_t callback,
+    void *userdata);
```

Comment:
would it be safe to assume that `sentry_span_to_sentry_trace` and `sentry_transaction_from_sentry_trace` will be implemented in a different PR?

Reply:
I wouldn’t implement those at all, because:
(https://develop.sentry.dev/sdk/performance/#new-span-and-transaction-classes)

> This should be preferred by integration/framework-sdk authors over fromSentryTrace as it hides the exact header names used deeper in the core sdk, and leaves opportunity for using additional headers (from the W3C) in the future without changing all integrations.